share/spack/spack-completion.bash

```diff
@@ -1282,7 +1283,7 @@ function _all_resource_hashes {
 }
 
 function _installed_packages {
-    spack --color=never find | grep -v "^--"
+    spack --color=never find --no-sections | grep -v "^--"
```

Comment:
```suggestion
    spack --color=never find --no-sections
```

Reply:
Good point.  Interestingly, the `grep` was ineffective for me when I tried this (and did not notice it).  I'm not sure why.

Reply:
That's because it's a `grep -v`, so it ignores all lines containing `--`. It's really just redundant.

Reply:
No, I know that -- I mean that I was seeing the `--` and other stuff in my tab completion output.

Reply:
Ahh, I see `==> 322 installed packages` but not `--`. I don't think `==> ...` used to pipe, but now it does.

Reply:
ok this is fixed.